### PR TITLE
Better error messages when pipelines reference incompatible aggs

### LINKED ISSUE #25273: Sum bucket aggregation cannot sum more than one level deep
Executing a sum_bucket aggregation with a bucket path multiple levels deep, like `bucket_1>bucket_2>metric` will fail with:

```
{
  "error": {
    "root_cause": [],
    "type": "search_phase_execution_exception",
    "reason": "",
    "phase": "fetch",
    "grouped": true,
    "failed_shards": [],
    "caused_by": {
      "type": "aggregation_execution_exception",
      "reason": "buckets_path must reference either a number value or a single value numeric metric aggregation, got: java.lang.Object[]"
    }
  },
  "status": 503
}
```

It would be preferable that this work. You can work around this by having one `sum_bucket` per aggregation depth level, but it would be nice to either have this return a more relevant error or just work.

### PR DESCRIPTION
**Pull request by polyfractal**
This is basically a less-invasive reboot of https://github.com/elastic/elasticsearch/pull/30152.

Pipeline aggs need a numeric value, or a single-valued agg to work properly.  If they reference multi-valued aggs or anything else, they throw a rather cryptic error.  This PR, like #30152, attempts to make the error nicer.

Unlike the previous attempt, this is much less invasive.  It tries to see if the current problematic object is a `MultiValue` agg and then provides some helpful text.  Otherwise, it throws an exception like before but now includes the aggregation name so the user can help determine which "level" of the path went bad.

Closes https://github.com/elastic/elasticsearch/issues/25273

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketHelpers.java [hunk 1]
```diff
@@ -156,6 +156,8 @@ public static Double resolveBucketValue(MultiBucketsAggregation agg,
             InternalMultiBucketAggregation.InternalBucket bucket, List<String> aggPathAsList, GapPolicy gapPolicy) {
         try {
             Object propertyValue = bucket.getProperty(agg.getName(), aggPathAsList);
+
+
```
Comment (iverase) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketHelpers.java]: nit: extra space

**Review by iverase — APPROVED** [review]
LGTM

**Comment by polyfractal** [comment]
<3

**Comment by polyfractal** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample please

**Pull request merged**